resilience4j-spring-boot2/src/main/java/io/github/resilience4j/circuitbreaker/monitoring/endpoint/CircuitBreakerServerSideEvent.java

```diff
@@ -53,59 +58,46 @@ public CircuitBreakerServerSideEvent(
-                .id(cbEvent.getCircuitBreakerName())
-                .event(cbEvent.getEventType().name())
-                .data(data.apply(cbEvent))
-                .build()
-        );
+        return Flux.merge(publishEvents(Array.ofAll(eventStreams)),getHeartbeatStream());
     }
 
-    @ReadOperation(produces = "text/event-stream")
-    public Flux<ServerSentEvent<String>> getEventsFilteredByCircuitBreakerName(
-        @Selector String name) {
-        CircuitBreaker givenCircuitBreaker = getCircuitBreaker(name);
-        Seq<Flux<CircuitBreakerEvent>> eventStreams = circuitBreakerRegistry.getAllCircuitBreakers()
-            .filter(circuitBreaker -> circuitBreaker.getName().equals(givenCircuitBreaker.getName()))
-            .map(
-                circuitBreaker -> toFlux(circuitBreaker.getEventPublisher())
-            );
-        Function<CircuitBreakerEvent, String> data = getCircuitBreakerEventStringFunction();
-        return Flux.merge(eventStreams).map(
-            cbEvent -> ServerSentEvent.<String>builder()
-                .id(cbEvent.getCircuitBreakerName())
-                .event(cbEvent.getEventType().name())
-                .data(data.apply(cbEvent))
-                .build()
-        );
+    @ReadOperation(produces = TEXT_EVENT_STREAM_VALUE)
+    public Flux<ServerSentEvent<String>> getEventsFilteredByCircuitBreakerName(@Selector String name) {
+        CircuitBreaker circuitBreaker = getCircuitBreaker(name);
+        Flux<CircuitBreakerEvent> eventStream = toFlux(circuitBreaker.getEventPublisher());
+        return Flux.merge(publishEvents(Array.of(eventStream)), getHeartbeatStream());
     }
 
-    @ReadOperation(produces = "text/event-stream")
+
+    @ReadOperation(produces = TEXT_EVENT_STREAM_VALUE)
     public Flux<ServerSentEvent<String>> getEventsFilteredByCircuitBreakerNameAndEventType(
         @Selector String name, @Selector String eventType) {
-        CircuitBreaker givenCircuitBreaker = getCircuitBreaker(name);
-        Seq<Flux<CircuitBreakerEvent>> eventStream = circuitBreakerRegistry.getAllCircuitBreakers()
-            .filter(circuitBreaker -> circuitBreaker.getName().equals(givenCircuitBreaker.getName()))
-            .map(
-                circuitBreaker -> toFlux(circuitBreaker.getEventPublisher())
+        CircuitBreaker circuitBreaker = getCircuitBreaker(name);
+        Flux<CircuitBreakerEvent> eventStream = toFlux(circuitBreaker.getEventPublisher())
+            .filter(
+                event -> event.getEventType() == CircuitBreakerEvent.Type.valueOf(eventType.toUpperCase())
+            );
+        return Flux.merge(publishEvents(Array.of(eventStream)), getHeartbeatStream());
+    }
+
+    private Flux<ServerSentEvent<String>> publishEvents(Seq<Flux<CircuitBreakerEvent>> eventStreams) {
+        Function<CircuitBreakerEvent, String> circuitBreakerEventDataFn = getCircuitBreakerEventStringFunction();
+        return Flux.merge(eventStreams)
+            .onBackpressureDrop()
```

Comment:
onBackpressureDrop && delayElements is needed in case of consumer is not keeping up with producer